src/Plugins/PluginAutoloadDependencies.php

```diff
@@ -10,9 +10,15 @@
 use Symfony\Component\Console\Input\InputInterface;
 
 /**
- * Class PluginAutoload
+ * Class PluginAutoloadDependencies
+ *
+ * Autoload the dependencies of a plugin -- libraries in the `require` section
+ * of the plugin's composer.json file.
+ *
+ * Note that an autoloader for the plugin's classes itself is registered
+ * by the PluginInfo class at plugin load time.
  */
-class PluginAutoload implements InitializeHookInterface, LoggerAwareInterface
+class PluginAutoloadDependencies implements InitializeHookInterface, LoggerAwareInterface
```

Comment:
I renamed this class to make it clear that now, this class is only responsible for autoloading the classes in the **dependencies** of the plugin. The classes defined directly in the plugin's autload section in its composer.json file are autoloaded at plugin discovery time, right after validation succeeds.